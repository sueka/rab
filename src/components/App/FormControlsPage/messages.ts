import { defineMessages } from 'react-intl'

export default defineMessages({
  formControls: 'Form controls',
})
