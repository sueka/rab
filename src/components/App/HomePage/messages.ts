import { defineMessages } from 'react-intl'

export default defineMessages({
  helloWorld: 'Hello, world!',
})
