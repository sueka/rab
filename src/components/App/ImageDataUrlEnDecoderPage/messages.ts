import { defineMessages } from 'react-intl'

export default defineMessages({
  imageDataUrlEnDecoder: 'Imageâ€“data URL en-/decoder',
  encoder: 'Encoder',
  decoder: 'Decoder',
})
