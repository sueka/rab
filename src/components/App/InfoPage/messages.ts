import { defineMessages } from 'react-intl'

export default defineMessages({
  info: 'Info',
})
