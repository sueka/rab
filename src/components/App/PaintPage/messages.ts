import { defineMessages } from 'react-intl'

export default defineMessages({
  paint: 'paint',
  clear: 'clear',
})
