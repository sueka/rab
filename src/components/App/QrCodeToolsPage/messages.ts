import { defineMessages } from 'react-intl'

export default defineMessages({
  qrCodeTools: 'QR code tools',
  encoder: 'Encoder',
  decoder: 'Decoder',
})
