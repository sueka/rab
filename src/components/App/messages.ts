import { defineMessages } from 'react-intl'

export default defineMessages({
  home: 'home',
  chess: 'chess',
  counter: 'counter',
  info: 'info',
  reminder: 'reminder',
})
