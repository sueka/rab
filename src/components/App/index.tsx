import { hot } from 'react-hot-loader'
import React from 'react'
import { Switch } from 'react-router'
import Helmet from 'react-helmet'
import { InjectedIntlProps, injectIntl } from 'react-intl'

import CssBaseline from '@material-ui/core/CssBaseline'

import Route from '~/components/Route'

import messages from './messages'

export /* for testing */ const HomePage = React.lazy(() => import(/* webpackChunkName: "homePage" */ '~/components/HomePage'))
export /* for testing */ const CounterPage = React.lazy(() => import(/* webpackChunkName: "counter" */ '~/components/CounterPage'))
export /* for testing */ const InfoPage = React.lazy(() => import(/* webpackChunkName: "info" */ '~/components/InfoPage'))
export /* for testing */ const ReminderPage = React.lazy(() => import(/* webpackChunkName: "reminder" */ '~/components/ReminderPage'))
export /* for testing */ const NoMatch = React.lazy(() => import(/* webpackChunkName: "noMatch" */ '~/components/NoMatch'))

type Props =
  & InjectedIntlProps

const App: React.FunctionComponent<Props> = ({ intl: { formatMessage } }) => (
  <>
    <CssBaseline />
    <Helmet
      titleTemplate="%s - react-app-prototype"
      defaultTitle="react-app-prototype"
    />
    <Switch>
      <Route path="/" component={ HomePage } helmetProps={ { title: formatMessage(messages.home) } } />
      <Route path="/counter" component={ CounterPage } helmetProps={ { title: formatMessage(messages.counter) } } />
      <Route path="/info" component={ InfoPage } helmetProps={ { title: formatMessage(messages.info) } } />
      <Route path="/reminder" component={ ReminderPage } helmetProps={ { title: formatMessage(messages.reminder) } } />
      <Route path="*" component={ NoMatch } />
    </Switch>
  </>
)

export default hot(module)(injectIntl(App))
