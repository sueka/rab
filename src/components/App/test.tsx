import React from 'react'
import { Provider } from 'react-redux'
import { StaticRouter } from 'react-router'
import { DndProvider } from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'
import { render } from '@testing-library/react'
import createMockStore from 'redux-mock-store'
import { SnackbarProvider } from 'notistack'

import typed from '~/lib/typed'
import { State } from '~/redux'
import IntlProvider from '~/components/IntlProvider'
import App, { HomePage, CounterPage, InfoPage, ReminderPage } from '.'
import formats from '../../../public/formats/en.json' // tslint:disable-line:no-relative-imports
import messages from '../../../public/messages/en.json' // tslint:disable-line:no-relative-imports

// NOTE: connected-react-router ではないので router state は不要。
const store = createMockStore<Alt.Omit<State, 'router'>>()({
  chess: {
    board: {
      pieces: [],
    },
  },
  counter: {
    count: 0,
  },
  io: {
    now: new Date('2019-07-27'),
  },
  localeSelector: {
    availableLocales: [
      'en',
      'ja',
    ],
    locale: 'en',
    formats,
    messages,
    errors: [],
  },
  reminder: {
    tasks: [],
    errors: {},
  },
})

describe.each`
location
${ '/' }
${ '/counter' }
${ '/info' }
${ '/reminder' }
${ '/nonexistent-path' }
`('App', ({ location }: { location: string }) => {
  test(typed<[string]>`at ${ location }`, async () => {
    const context = {}

    const { container } = render(
      <Provider store={ store }>
        <IntlProvider>
          <DndProvider backend={ TestBackend }>
            <StaticRouter context={ context } location={ location }>
              <SnackbarProvider>
                <App />
              </SnackbarProvider>
            </StaticRouter>
          </DndProvider>
        </IntlProvider>
      </Provider>
    )

    await HomePage
    await CounterPage
    await InfoPage
    await ReminderPage

    expect(container).toMatchSnapshot()
  })
})
