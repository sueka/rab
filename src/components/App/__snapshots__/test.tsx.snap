// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App at / 1`] = `
<div>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="stub uuid"
    >
      Languages
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        data-testid="localeSelect"
        id="stub uuid"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="ja"
        >
          日本語
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/counter"
  >
    counter
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/info"
  >
    info
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/reminder"
  >
    reminder
  </a>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Hello, world!
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    $100.00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Jul 27, 2019
  </p>
  Error: No route matches /.
</div>
`;

exports[`App at /counter 1`] = `
<div>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="stub uuid"
    >
      Languages
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        data-testid="localeSelect"
        id="stub uuid"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="ja"
        >
          日本語
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/counter"
  >
    counter
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/info"
  >
    info
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/reminder"
  >
    reminder
  </a>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Hello, world!
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    $100.00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Jul 27, 2019
  </p>
  <div>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      0
    </span>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        reset
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        +
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        -
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="incrementIfOddButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + if odd
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="incrementAsyncButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + in 1 second
      </span>
    </button>
  </div>
</div>
`;

exports[`App at /info 1`] = `
<div>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="stub uuid"
    >
      Languages
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        data-testid="localeSelect"
        id="stub uuid"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="ja"
        >
          日本語
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/counter"
  >
    counter
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/info"
  >
    info
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/reminder"
  >
    reminder
  </a>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Hello, world!
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    $100.00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Jul 27, 2019
  </p>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Fetch data
    </span>
  </button>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Fetching not started.
  </p>
</div>
`;

exports[`App at /reminder 1`] = `
<div>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="stub uuid"
    >
      Languages
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        data-testid="localeSelect"
        id="stub uuid"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="ja"
        >
          日本語
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/counter"
  >
    counter
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/info"
  >
    info
  </a>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/reminder"
  >
    reminder
  </a>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Hello, world!
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    $100.00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Jul 27, 2019
  </p>
  <ul
    class="MuiList-root MuiList-padding"
  />
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      add
    </span>
  </button>
</div>
`;
